staging/src/k8s.io/apimachinery/pkg/api/errors/errors_test.go

```diff
@@ -220,3 +220,39 @@ func TestFromObject(t *testing.T) {
 		}
 	}
 }
+
+func TestReasonForErrorSupportsWrappedErrors(t *testing.T) {
+	testCases := []struct {
+		name                string
+		err                 error
+		expectedKnownReason bool
```

Comment:
Wouldn't it be better to actually match the wrapped `Reason` rather than just a boolean?